### PR TITLE
DSD-1386 & DSD-1388: new grays and better dark mode color values

### PR DESCRIPTION
**Pull request by bigfishdesign13**
Fixes JIRA ticket [DSD-1386](https://jira.nypl.org/browse/DSD-1386) & [DSD-1388](https://jira.nypl.org/browse/DSD-1388)

## This PR does the following:

- Adds `ui.gray.semi-medium` and `ui.gray.semi-dark` to the color palette.
- Updates the `dark.ui.border.default` design token to use `ui.gray.semi-dark` as its base.
- Updates the `dark.ui.border.hover` design token to use `ui.gray.semi-medium` as its base.
- Updates the `dark.ui.disabled.primary` design token to use `ui.gray.dark` as its base.
- Updates the `dark.ui.disabled.secondary` design token to use `ui.gray.x-dark` as its base.
- Updates the `Toggle` component to use `ui.gray.semi-dark` color for its background color.
- Updates the `List` and `StyledList` components to use `ui.gray.semi-dark` for the bullet color in the "ul" variant.
- Updates the `Slider` component to use `ui.bg.active` for the color of the track bar in the "disabled" state.
- Fixes a `border-radius` formatting issues on the hover state of the search field for the dark mode version of the `Header` component.
- Fixes a `color` issue on the hover state of the links for the dark mode version of the `Footer` component.
- Fixes a `border-color` issue for the dark mode version of the `FeedbackBox` component.

**NOTE:** Some additional updates are sprinkled throughout this update to cleanup and improve dark mode color values.

## How has this been tested?

- local Storybook

## Accessibility concerns or updates

- The new gray colors were added to give better options to meet color contrast compliance.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,20 @@ Currently, this repo is in Prerelease. When it is released, this project will ad
...
+
+### Updates
+
+- Updates the `dark.ui.border.default` design token to use `ui.gray.semi-dark` as its base.
+- Updates the `dark.ui.border.hover` design token to use `ui.gray.semi-medium` as its base.
+- Updates the `dark.ui.disabled.primary` design token to use `ui.gray.dark` as its base.
+- Updates the `dark.ui.disabled.secondary` design token to use `ui.gray.x-dark` as its base.
+- Updates the `Toggle` component to use `ui.gray.semi-dark` color for its background color.
+- Updates the `List` and `StyledList` components to use `ui.gray.semi-dark` for the bullet color in the "ul" variant.
+- Updates the `Slider` component to use `ui.bg.active` for the color of the track bar in the "disabled" state.
```
Comment (EdwinGuzman) [inline CHANGELOG.md:23]: Should add a line that `DatePicker` was updated as well.

Comment (EdwinGuzman) [reply CHANGELOG.md:23]: Should `FeedbackBox` also be added here? Maybe even Header/Footer?

**Review by EdwinGuzman — COMMENTED** [review]
The update looks good but one question, should the two new `semi-` colors be added in the "Colors" doc?

File: CHANGELOG.md (hunk 1 above)
Comment (bigfishdesign13) [reply CHANGELOG.md:23]: The changes were so minimal, I wasn't sure if they should be added.  However, to be thorough and consistent, I will include something in the CHANGELOG.

**Comment by bigfishdesign13** [comment]
> The update looks good but one question, should the two new `semi-` colors be added in the "Colors" doc?

@EdwinGuzman Yes!  Good catch.  I thought about this while I was working, but then I got antsy and pushed the PR before I took care of it.

**Review by EdwinGuzman — APPROVED** [review]
Update looks good. I think a separate update can be done to the "Design Tokens" doc page.

**Review by EdwinGuzman — APPROVED** [review]

**Pull request merged**